chgnet/data/dataset.py

```diff
@@ -24,24 +25,24 @@ class StructureData(Dataset):
 
     def __init__(
         self,
-        structures: list,
-        energies: list,
-        forces: list,
-        stresses: list = None,
-        magmoms: list = None,
+        structures: list[dict[str, Any]],
+        energies: list[float],
+        forces: list[Sequence[Sequence[float]]],
+        stresses: list[Sequence[Sequence[float]]] = None,
+        magmoms: list[Sequence[Sequence[float]]] = None,
         graph_converter: CrystalGraphConverter = None,
     ) -> None:
         """Initialize the dataset.
 
         Args:
-            structures (list): a list of structures
-            energies (list): a list of  energies
-            forces (list): a list of forces
-            stresses (List, optional): a list of stresses
-            magmoms (List, optional): a list of magmoms
-            graph_converter (CrystalGraphConverter, optional):
-                a CrystalGraphConverter to convert the structures,
-                if None, it will be set to CHGNet default converter
+            structures (list[dict]): a list of structures in pymatgen dict format, i.e.
+                the output of Structure.to_dict()
+            energies (list[float]): [data_size, 1]
+            forces (list[list[float]]): [data_size, n_atoms, 3]
+            stresses (list[list[float]], optional): [data_size, 3, 3]
+            magmoms (list[list[float]], optional): [data_size, n_atoms, 3]
+            graph_converter (CrystalGraphConverter, optional): Converts the structures to
+                graphs. If None, it will be set to CHGNet default converter.
```

Comment:
Are these shapes correct?

Reply:
magmoms should be [data_size, n_atoms, 1]

Reply:
Fixed.